setup_utils.py

```diff
@@ -0,0 +1,44 @@
+import os
+import re
+
+
+def update_pins(setup_args):
+    # Use requirements and constraints to set version pins
+    packages = set()
+    with open('./requirements.txt') as requirements:
+        for r in requirements:
+            if r.lower().strip() == 'dallinger':
+                continue
+            if not r.startswith('-') or r.startswith('#'):
+                packages.add(r.strip().lower())
+
+    requirements = []
+    REQUIREMENT_RE = re.compile(r'^(([^=]+)==[^#]+)(#.*)?$')
```

Comment:
Not presently. Do we anticipate using ranges in `constraints.txt`? Because we're letting Pyup manage that file, I think it probably makes sense to assume fixed versions. It would probably be sufficient to change that to`[<>=]+` though